### PR TITLE
Add MonthlyBudget tests

### LINKED ISSUE #115: Bring up code coverage
Code coverage is currently at 69.29%. The aim is to bring this up to 100%.

### PR DESCRIPTION
**Pull request by zhaojj2209**
Partially resolves #115.

Added `MonthlyBudgetTest` class to test methods in `MonthlyBudget`.

Also removed traces of `UndoCommand` and `RedoCommand` from javadocs.

### CODE REVIEW

**Review by wltan — COMMENTED** [review]
Pitest: https://ay2021s1-cs2103t-w16-3.github.io/reports/pitest/202011071714/ay2021s1_cs2103_w16_3.finesse.model.budget/MonthlyBudget.java.html

Line coverage is there. Is it possible to increase mutation coverage as well?

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/model/budget/MonthlyBudgetTest.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.Collections;
+
+import org.junit.jupiter.api.Test;
+
+import ay2021s1_cs2103_w16_3.finesse.model.FinanceTracker;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Amount;
+import ay2021s1_cs2103_w16_3.finesse.model.transaction.Amount.CalculatedAmount;
+
+public class MonthlyBudgetTest {
+    private static final int NUM_OF_MONTHS = 12;
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/model/budget/MonthlyBudgetTest.java:18]:
```suggestion
```
Redundant field.

**Review by yongping827 — CHANGES_REQUESTED** [review]

File: src/main/java/ay2021s1_cs2103_w16_3/finesse/model/budget/MonthlyBudget.java [hunk 2]
```diff
@@ -193,6 +193,13 @@ private void calculateSavings() {
...
                 || (other instanceof MonthlyBudget // instanceof handles nulls
-                && monthlyExpenseLimit.equals(((MonthlyBudget) other).monthlyExpenseLimit));
+                && monthlyExpenseLimit.getValue().equals(((MonthlyBudget) other).monthlyExpenseLimit.getValue())
+                && monthlySavingsGoal.getValue().equals(((MonthlyBudget) other).monthlySavingsGoal.getValue())
+                && remainingBudget.getValue().equals(((MonthlyBudget) other).remainingBudget.getValue())
+                && currentSavings.getValue().equals(((MonthlyBudget) other).currentSavings.getValue())
+                && monthlyExpenses.equals(((MonthlyBudget) other).monthlyExpenses)
+                && monthlyIncomes.equals(((MonthlyBudget) other).monthlyIncomes)
+                && monthlySavings.equals(((MonthlyBudget) other).monthlySavings)
+                && months.equals(((MonthlyBudget) other).months));
```
Comment (ianyong) [inline src/main/java/ay2021s1_cs2103_w16_3/finesse/model/budget/MonthlyBudget.java:203]: If code duplication is a concern, might be worth rewriting this such that you cast `other` to type `MonthlyBudget` only once.

**Review by wltan — APPROVED** [review]
LGTM!

**Review by ianyong — APPROVED** [review]
LGTM!

**Review by yongping827 — APPROVED** [review]
LGTM

**Pull request merged**